test/unit/test_dss_client.py

```diff
@@ -335,6 +337,20 @@ def _assert_bundle_json(self, prefix=''):
             actual_hash = hashlib.sha256(f.read()).hexdigest()
         self.assertEqual(actual_hash, expected_hash)
 
+    @unittest.skipIf(os.name is 'nt', 'os.link is not used on Windows')
+    def test_link_fails(self):
+        """If os.link fails on *nix, we should revert to copy"""
+        os_error = OSError()
+        os_error.errno = errno.EMLINK
+        with patch('os.link', side_effect=os_error):
+            from hca.dss.util import log as log_
+            with self.assertLogs(logger=log_, level=logging.WARNING) as logs:
+                self._mock_download_manifest(self.manifest_file, 'aws', layout='bundle')
+        for record in logs.records:
+            self.assertEqual(record.funcName, 'hardlink')
+            self.assertEqual(record.msg, 'Too many hard links to a file, reverting to copying')
+        self.assertEqual(len(logs.records), 9)
```

Comment:
Indent this line so it's under the ContextManager at line 348 ending in `as logs:`.

Reply:
I don't want it under the context manager, because then if that assert statement logs anything, it will change the number of records. Only things that I want to capture the number of records for should be under the context manager.

Reply:
It makes sense to minimize the bodies of `with` statements. What would moving the assert into the logs CM buy us?

Reply:
I'd document why 9 and what it depends on.